client/src/components/game/menu/HeaderBar.vue

```diff
@@ -46,9 +46,9 @@
                     <button class="btn btn-primary btn-sm mr-1 mb-1" @click="fitGalaxy" title="Hotkey: 0"><i class="fas fa-compass"></i></button>
                     <button class="btn btn-primary btn-sm mr-1 mb-1" @click="zoomIn()" title="Hotkey: +"><i class="fas fa-search-plus"></i></button>
                     <button class="btn btn-primary btn-sm mr-1 mb-1" @click="zoomOut()" title="Hotkey: -"><i class="fas fa-search-minus"></i></button>
-                    <button class="btn btn-primary btn-sm mr-1 mb-1" @click="setMenuState(MENU_STATES.COMBAT_CALCULATOR)" title="Hotkey: C"><i class="fas fa-calculator"></i></button>
-                    <div v-if="userPlayer">
-                        <button class="btn btn-primary btn-sm mr-1 mb-1" @click="panToHomeStar()" title="Hotkey: H"><i class="fas fa-home"></i></button>
+                    <button class="btn btn-primary btn-sm mr-1 mb-1" @click="panToHomeStar()" title="Hotkey: H"><i class="fas fa-home"></i></button>
+                  <div v-if="userPlayer">
```

Comment:
This is going to cause a problem. The `userPlayer` object determines whether the player has joined the game, if its `null` then the player isn't in the game yet. I can see that you have moved around the buttons correctly however the visibility logic isn't quite right.

- The player **should not** be able to click on the "Home" button if `userPlayer` object is null as they aren't in the game.
- The player **should** be able to click on the calculator button **always**, even if they aren't in the game.

This should be pretty simple to fix, put the `v-if="userPlayer"` on the home, ruler and bulk upgrade buttons and remove the `v-if` on this `div`. I think that should resolve it.

Reply:
Hi @mike-eason , thank you for the detailed feedback. Just pushed that suggested fixed 😀.